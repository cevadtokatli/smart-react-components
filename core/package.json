{
  "name": "@smart-react-components/core",
  "version": "2.0.0-beta-0",
  "license": "MIT",
  "main": "index.js",
  "description": "SRC core includes the theme, Styled properties, and Element components.",
  "keywords": [
    "javascript",
    "typescript",
    "react",
    "styled-components"
  ],
  "author": {
    "name": "Cevad Tokatlı",
    "email": "cevadtokatli@hotmail.com"
  },
  "contributors": [
    {
      "name": "Cemil Tokatlı",
      "email": "cemil@tokatli.me"
    }
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/cevadtokatli/smart-react-components/tree/v2/core"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "yarn clear && tsc --outDir lib && yarn copy",
    "clear": "rm -rf lib",
    "copy": "cp package.json lib/package.json && cp README.md lib/README.md && cp ../LICENSE lib/LICENSE",
    "dev": "tsc -w --outDir lib",
    "lint": "eslint src",
    "release": "../scripts/release.sh core",
    "test": "jest"
  },
  "devDependencies": {
    "@types/color": "3.0.3",
    "@types/enzyme": "3.10.12",
    "@types/jest": "29.0.1",
    "@types/react": "18.0.19",
    "@types/react-dom": "18.0.6",
    "@types/styled-components": "5.1.26",
    "@wojtekmaj/enzyme-adapter-react-17": "0.6.7",
    "enzyme": "3.11.0",
    "eslint": "8.0.1",
    "jest": "29.0.3",
    "jsdom": "20.0.0",
    "jsdom-global": "3.0.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "styled-components": "5.3.5",
    "ts-jest": "29.0.0",
    "typescript": "4.8.3"
  },
  "dependencies": {
    "color": "4.2.3"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "styled-components": "^5.3.5"
  }
}
