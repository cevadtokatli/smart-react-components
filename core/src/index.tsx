export * from './theme'
export { default as createTheme } from './theme'
export * from './styled-props'
export * from './Element'
