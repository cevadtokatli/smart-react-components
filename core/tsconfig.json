{
  "extends": "../tsconfig.json",
  "include": ["src/**/*.tsx"],
  "exclude": ["node_modules", "src/**/*.test.tsx"]
}
