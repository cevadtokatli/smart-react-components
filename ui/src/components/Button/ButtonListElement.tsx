import Div from '@smart-react-components/core/Element/Div'
import { StyledProps } from '@smart-react-components/core/styled-props'
import { Partial, ResponsiveProp, SizeProp } from '@smart-react-components/core/types'
import styled from 'styled-components'

export interface Props extends
  StyledProps,
  Partial<ResponsiveProp<'buttonSize', SizeProp>> {
  hasSpace?: boolean
  isBlock?: boolean
  length?: number
}

export default styled(Div).attrs<Props>(({ hasSpace }: Props) => ({
  getButtonSize: (v, t) => `
    ${hasSpace
      ? `
        margin: -${t.$.size.button[v].margin.y} -${t.$.size.button[v].margin.x};
      `
      : ''
    }
  `,
}))<Props>(({ isBlock, length }: Props) => `
  align-items: center;
  display: flex;
  justify-content: flex-start;

  ${isBlock
    ? `
      > * {
        flex: 1 1 ${100 / length}%;
        width: ${100 / length}%;
      }
    `
    : `
      flex-wrap: wrap;
    `
  }
`)
