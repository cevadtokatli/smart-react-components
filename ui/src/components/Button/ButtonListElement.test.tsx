import { wrapTheme } from '@smart-react-components/core/test'
import { render } from '@testing-library/react'
import React from 'react'
import ButtonListElement from './ButtonListElement'

describe('<ButtonListElement />', () => {
  let props

  beforeAll(() => {
    props = {
      buttonSize: 'medium',
      hasSpace: false,
      isBlock: false,
      length: 10,
    }
  })

  it('should render component', () => {
    const { asFragment } = render(wrapTheme(<ButtonListElement {...props} />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('children should have space', () => {
    const { asFragment } = render(wrapTheme(<ButtonListElement {...props} hasSpace />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('children should be block', () => {
    const { asFragment } = render(wrapTheme(<ButtonListElement {...props} isBlock />))
    expect(asFragment()).toMatchSnapshot()
  })
})
