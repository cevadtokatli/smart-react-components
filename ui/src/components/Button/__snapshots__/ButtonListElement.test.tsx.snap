// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonListElement /> children should be block 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 > * {
  -webkit-flex: 1 1 10%;
  -ms-flex: 1 1 10%;
  flex: 1 1 10%;
  width: 10%;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`<ButtonListElement /> children should have space 1`] = `
<DocumentFragment>
  .c0 {
  margin: -.5rem -.5rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
    class="c0 c1"
  />
</DocumentFragment>
`;

exports[`<ButtonListElement /> should render component 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
